raft/src/main/java/org/apache/kafka/raft/FileBasedStateStore.java

```diff
@@ -149,6 +149,7 @@ private void writeElectionStateToFile(final File stateFile, QuorumStateData stat
             writer.write(jsonState.toString());
             writer.flush();
             fileOutputStream.getFD().sync();
+            fileOutputStream.close();
```

Comment:
I checked the code and it is safe to close twice. I'm thinking maybe we add a comment above this line to describe why we explicitly close here even though we already use try with resource. I think we should still keep the try with resource to make sure it'll get closed in the end. WDYT?